src/libutil/strutil_test.cpp

```diff
@@ -353,41 +354,123 @@ void test_replace ()
-    OIIO_CHECK_EQUAL (Strutil::from_string<int>(" 123 "), 123);
-    OIIO_CHECK_EQUAL (Strutil::from_string<int>("123.45"), 123);
-
-    OIIO_CHECK_EQUAL (Strutil::from_string<unsigned int>("hi"), unsigned(0));
-    OIIO_CHECK_EQUAL (Strutil::from_string<unsigned int>("123"), unsigned(123));
-    OIIO_CHECK_EQUAL (Strutil::from_string<unsigned int>("-123"), unsigned(-123));
-    OIIO_CHECK_EQUAL (Strutil::from_string<unsigned int>(" 123 "), unsigned(123));
-    OIIO_CHECK_EQUAL (Strutil::from_string<unsigned int>("123.45"), unsigned(123));
-
-    OIIO_CHECK_EQUAL (Strutil::from_string<float>("hi"), 0.0f);
-    OIIO_CHECK_EQUAL (Strutil::from_string<float>("123"), 123.0f);
-    OIIO_CHECK_EQUAL (Strutil::from_string<float>("-123"), -123.0f);
-    OIIO_CHECK_EQUAL (Strutil::from_string<float>("123.45"), 123.45f);
-    OIIO_CHECK_EQUAL (Strutil::from_string<float>(" 123.45 "), 123.45f);
-    OIIO_CHECK_EQUAL (Strutil::from_string<float>("123.45+12"), 123.45f);
-    OIIO_CHECK_EQUAL (Strutil::from_string<float>("1.2345e+2"), 123.45f);
+    size_t pos;
+
+    OIIO_CHECK_EQUAL (Strutil::string_is_int("142"), true);
+    OIIO_CHECK_EQUAL (Strutil::string_is_int("-142"), true);
+    OIIO_CHECK_EQUAL (Strutil::string_is_int("+142"), true);
+    OIIO_CHECK_EQUAL (Strutil::string_is_int("142.0"), false);
+    OIIO_CHECK_EQUAL (Strutil::string_is_int(""), false);
+    OIIO_CHECK_EQUAL (Strutil::string_is_int("  "), false);
+    OIIO_CHECK_EQUAL (Strutil::string_is_int("foo"), false);
+    OIIO_CHECK_EQUAL (Strutil::string_is_int("142x"), false);
+    OIIO_CHECK_EQUAL (Strutil::string_is_int(" 142"), true);
+    OIIO_CHECK_EQUAL (Strutil::string_is_int("142 "), true);
+    OIIO_CHECK_EQUAL (Strutil::string_is_int("x142"), false);
+
+    OIIO_CHECK_EQUAL (Strutil::string_is_float("142"), true);
+    OIIO_CHECK_EQUAL (Strutil::string_is_float("142.0"), true);
+    OIIO_CHECK_EQUAL (Strutil::string_is_float(""), false);
+    OIIO_CHECK_EQUAL (Strutil::string_is_float("  "), false);
+    OIIO_CHECK_EQUAL (Strutil::string_is_float("foo"), false);
+    OIIO_CHECK_EQUAL (Strutil::string_is_float("142x"), false);
+    OIIO_CHECK_EQUAL (Strutil::string_is_float(" 142"), true);
+    OIIO_CHECK_EQUAL (Strutil::string_is_float(" 142 "), true);
+    OIIO_CHECK_EQUAL (Strutil::string_is_float(" 142.0 "), true);
+    OIIO_CHECK_EQUAL (Strutil::string_is_float("x142"), false);
+
+    // Note: we don't test string_is<> separately because it's just
+    // implemented directly as calls to string_is_{int,float}.
+
+    OIIO_CHECK_EQUAL (Strutil::stoi("hi"), 0);
+    OIIO_CHECK_EQUAL (Strutil::stoi("  "), 0);
+    OIIO_CHECK_EQUAL (Strutil::stoi("123"), 123);
+    OIIO_CHECK_EQUAL (Strutil::stoi("-123"), -123);
+    OIIO_CHECK_EQUAL (Strutil::stoi("+123"), 123);
+    OIIO_CHECK_EQUAL (Strutil::stoi(" 123 "), 123);
+    OIIO_CHECK_EQUAL (Strutil::stoi("123.45"), 123);
+    OIIO_CHECK_EQUAL (Strutil::stoi("12345678901234567890"), std::numeric_limits<int>::max());
```

Comment:
Maybe add some overflow cases too?

Reply:
That is an overflow case on line 392. What did you mean?